Samples/excel-add-in-shape-dashboard/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,431 @@
+
+    let segment = sheet.getRange("E151:E151");
+    segment.formulas = [["=SUMPRODUCT(1/COUNTIF(C2:C98,C2:C98))"]];
+    let region = sheet.getRange("E152:E152");
+    region.formulas = [["=SUMPRODUCT(1/COUNTIF(D2:D98,D2:D98))"]];
+
+    let category = sheet.getRange("E153:E153");
+    category.formulas = [["=SUMPRODUCT(1/COUNTIF(E2:E98,E2:E98))"]];
+    let subCategory = sheet.getRange("E154:E154");
+    subCategory.formulas = [["=SUMPRODUCT(1/COUNTIF(F2:F98,F2:F98))"]];
+    let maxSale = sheet.getRange("E155:E155");
+    maxSale.formulas = [["=MAX(G2:G98)"]];
+    let sumSale = sheet.getRange("E156:E156");
+    sumSale.formulas = [["=SUM(G2:G98)"]];
+    let state = sheet.getRange("E157:E157");
+    state.formulas = [["=SUMPRODUCT(1/COUNTIF(H2:H98,H2:H98))"]];
+
+    await context.sync();
+    showStatus("Add sample data - success!", false);
+  });
+}
+
+async function createDashboard() {
+  await Excel.run(async (context) => {
+    showStatus("Running...", false);
+    let shapes = context.workbook.worksheets.getItem("Sample").shapes;
+    let array = [];
+
+    let region = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    region.left = 550;
+    region.top = 100;
+    region.width = 100;
+    region.height = 100;
+    array.push(region);
+
+    let state = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    state.left = 700;
+    state.top = 100;
+    state.width = 100;
+    state.height = 100;
+    array.push(state);
+
+    let line = shapes.addGeometricShape(Excel.GeometricShapeType.rectangle);
+    line.left = 500;
+    line.top = 230;
+    line.width = 350;
+    line.height = 1;
+    array.push(line);
+
+    let category = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    category.left = 550;
+    category.top = 260;
+    category.width = 100;
+    category.height = 100;
+    array.push(category);
+
+    let subCategory = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    subCategory.left = 700;
+    subCategory.top = 260;
+    subCategory.width = 100;
+    subCategory.height = 100;
+    array.push(subCategory);
+
+    let line2 = shapes.addGeometricShape(Excel.GeometricShapeType.rectangle);
+    line2.left = 500;
+    line2.top = 390;
+    line2.width = 350;
+    line2.height = 1;
+    array.push(line2);
+
+    let maxScale = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    maxScale.left = 550;
+    maxScale.top = 420;
+    maxScale.width = 100;
+    maxScale.height = 100;
+    array.push(maxScale);
+
+    let sumScale = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    sumScale.left = 700;
+    sumScale.top = 420;
+    sumScale.width = 100;
+    sumScale.height = 100;
+    array.push(sumScale);
+
+    shapes.addGroup(array);
+    await context.sync();
+
+    showStatus("Create empty dashboard - success!", false);
+  });
+}
+
+async function changeColor() {
```

Comment:
This method is doing way more than changing the color. It's also getting the data. I find that very confusing. Can we split the formatting code out from the data retrieval code?